src/index.ts

```diff
@@ -413,49 +396,47 @@ export class GalleryComponent extends BaseComponent {
       thumb.initialHeight = medianHeight;
     }
 
-    this._$thumbs.link($.templates.galleryThumbsTemplate, this._thumbs);
+    const renderedHtml = this._thumbs.map(this._galleryThumbsTemplate).join('');
+    this._$thumbs.html(renderedHtml);
 
     if (multiSelectState && !multiSelectState.isEnabled) {
       // add a selection click event to all thumbs
       this._$thumbs.delegate(".thumb", "click", function (e: any) {
         e.preventDefault();
-        const thumb = $.view(this).data;
-        that.fire(Events.THUMB_SELECTED, thumb);
+        const thumbIndex = parseInt($(this).attr("data-index") as string);
+        const thumb: MultiSelectableThumb = that._thumbs[thumbIndex];
+        if (thumb) {
+          that.fire(Events.THUMB_SELECTED, thumb);
+        }
       });
     } else {
       // make each thumb a checkboxButton
       const thumbs: JQuery = this._$thumbs.find(".thumb");
-
-      for (var i = 0; i < thumbs.length; i++) {
-        const that = this;
+    
+      for (let i = 0; i < thumbs.length; i++) {
         const $thumb = $(thumbs[i]);
-
-        $thumb.checkboxButton(function (_checked: boolean) {
-          const thumb: MultiSelectableThumb = $.view(<any>this).data;
-          that._setThumbMultiSelected(thumb, !thumb.multiSelected);
-          const range: MultiSelectableRange = <MultiSelectableRange>(
-            that.options.data.helper.getCanvasRange(thumb.data)
-          );
-          const multiSelectState: MultiSelectState | null =
-            that._getMultiSelectState();
-
-          if (multiSelectState) {
-            if (range) {
-              multiSelectState.selectRange(
-                <MultiSelectableRange>range,
-                thumb.multiSelected
-              );
-            } else {
-              multiSelectState.selectCanvas(
-                <MultiSelectableCanvas>thumb.data,
-                thumb.multiSelected
-              );
+    
+        $thumb.on("click", function () {
+          const thumbIndex = parseInt($(this).attr("data-index") as string);

```

Comment:
again no longer using views

Reply:
...and same suggestion as above, if it will work:

```suggestion
          const thumbIndex = parseInt(this.dataset.index as string);

```